doc/source/train/examples/intel_gaudi/llama_pretrain.ipynb

```diff
@@ -0,0 +1,568 @@
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class MegatronDataset:\n",
+    "    def __call__(self, config):\n",
+    "        def _train_valid_test_datasets_provider(train_val_test_num_samples):\n",
+    "            \"\"\"Build train, valid, and test datasets.\"\"\"\n",
+    "            args = get_args()\n",
+    "            print_rank_0(\"> building train, validation, and test datasets \" \"for GPT ...\")\n",
+    "            train_ds, valid_ds, test_ds = gpt_dataset.build_train_valid_test_datasets(\n",
+    "                data_prefix=args.data_path,\n",
+    "                data_impl=args.data_impl,\n",
+    "                splits_string=args.split,\n",
+    "                train_valid_test_num_samples=train_val_test_num_samples,\n",
+    "                seq_length=args.seq_length,\n",
+    "                seed=args.seed,\n",
+    "                skip_warmup=(not args.mmap_warmup),\n",
+    "                train_data_prefix=args.train_data_path,\n",
+    "                valid_data_prefix=args.valid_data_path,\n",
+    "                test_data_prefix=args.test_data_path,\n",
+    "                data_cache_path=args.data_cache_path,\n",
+    "            )\n",
+    "            print_rank_0(\"> finished creating GPT datasets ...\")\n",
+    "\n",
+    "            return train_ds, valid_ds, test_ds\n",
+    "\n",
+    "        args = get_args()\n",
+    "        update_train_iters(args)\n",
+    "        datasets = build_train_valid_test_datasets(_train_valid_test_datasets_provider)\n",
+    "        print_rank_0(datasets)\n",
+    "        return datasets\n",
+    "\n",
+    "\n",
+    "def load_datasets(config):\n",
+    "    dataset = MegatronDataset()\n",
+    "    return dataset(config)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Process dataset to dataloader"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class MegatronProcesser:\n",
+    "    def prepare(self, tokenizer, dataset, **kwargs):\n",
+    "        args = get_args()\n",
+    "\n",
+    "        (train_dataloader, valid_dataloader, test_dataloader) = (None, None, None)\n",
+    "\n",
+    "        print_rank_0(\"> building train, validation, and test datasets ...\")\n",
+    "        iteration = kwargs.get(\"step\", 0)\n",
+    "        if iteration:\n",
+    "            # passed value is starting step\n",
+    "            iteration -= 1\n",
+    "            args.consumed_train_samples = iteration * args.global_batch_size\n",
+    "            args.consumed_valid_samples = (\n",
+    "                (args.iteration // args.eval_interval) * args.eval_iters * args.global_batch_size\n",
+    "            )\n",
+    "\n",
+    "        # Data loader only on rank 0 of each model parallel group.\n",
+    "        if args.use_dataset_only or mpu.get_tensor_model_parallel_rank() == 0:\n",
```

Comment:
Where did we configure the tensor and pipeline parallel group size?

Reply:
tensor and pipeline parallel not supported yet, for this example no longer relies on megatron.